src/filters/local_rate_limit.rs

```diff
@@ -35,115 +36,143 @@ use self::quilkin::extensions::filters::local_rate_limit::v1alpha1::LocalRateLim
 pub const NAME: &str = "quilkin.extensions.filters.local_rate_limit.v1alpha1.LocalRateLimit";
 
 /// Creates a new factory for generating rate limiting filters.
-pub fn factory() -> DynFilterFactory {
-    Box::from(LocalRateLimitFactory)
+pub fn factory(base: &Logger) -> DynFilterFactory {
+    Box::from(LocalRateLimitFactory::new(base))
+}
+
+// TODO: we should make these values configurable and transparent to the filter.
+/// SESSION_TIMEOUT_SECONDS is the default session timeout.
+pub const SESSION_TIMEOUT_SECONDS: Duration = Duration::from_secs(60);
+
+/// SESSION_EXPIRY_POLL_INTERVAL is the default interval to check for expired sessions.
+const SESSION_EXPIRY_POLL_INTERVAL: Duration = Duration::from_secs(60);
+
+/// Bucket stores two atomics.
+/// - A counter that tracks how many packets we've processed within a time window.
+/// - A timestamp that stores the time we last reset the counter. It tracks
+///   the start of the time window.
+/// This allows us to have a simpler implementation for calculating token
+/// exhaustion without needing a write lock in the common case. The downside
+/// however is that since we're relying on two independent atomics, there is
+/// in theory, a chance that we could allow a few packets through (i.e in-between
+/// checking the counter and the timestamp). However, in practice this would be
```

Comment:
Added a section to call this out in the rate limiter doc at least. I imagine since this is UDP people aren't expecting any reliability guarantees from quilkin but we could explicitly call that out too